it/src/main/java/com/google/cloud/teleport/it/pubsub/DefaultPubsubResourceManager.java

```diff
@@ -0,0 +1,239 @@
+  private final CredentialsProvider credentialsProvider;
+  private final TopicAdminClient topicAdminClient;
+  private final SubscriptionAdminClient subscriptionAdminClient;
+
+  private final List<TopicName> createdTopics;
+  private final List<SubscriptionName> createdSubscriptions;
+
+  private DefaultPubsubResourceManager(Builder builder) throws IOException {
+    this(
+        builder.testName,
+        builder.projectId,
+        builder.credentialsProvider,
+        TopicAdminClient.create(
+            TopicAdminSettings.newBuilder()
+                .setCredentialsProvider(builder.credentialsProvider)
+                .build()),
+        SubscriptionAdminClient.create(
+            SubscriptionAdminSettings.newBuilder()
+                .setCredentialsProvider(builder.credentialsProvider)
+                .build()));
+  }
+
+  @VisibleForTesting
+  DefaultPubsubResourceManager(
+      String testName,
+      String projectId,
+      CredentialsProvider credentialsProvider,
+      TopicAdminClient topicAdminClient,
+      SubscriptionAdminClient subscriptionAdminClient) {
+    this.projectId = projectId;
+    this.testId = PubsubUtils.createTestId(testName);
+    this.credentialsProvider = credentialsProvider;
+    this.topicAdminClient = topicAdminClient;
+    this.subscriptionAdminClient = subscriptionAdminClient;
+    this.createdTopics = Collections.synchronizedList(new ArrayList<>());
+    this.createdSubscriptions = Collections.synchronizedList(new ArrayList<>());
+  }
+
+  public static Builder builder(String testName, String projectId) {
+    checkArgument(!Strings.isNullOrEmpty(testName), "testName can not be null or empty");
+    checkArgument(!projectId.isEmpty(), "projectId can not be empty");
+    return new Builder(testName, projectId);
+  }
+
+  @Override
+  public TopicName createTopic(String topicName) {
+    checkArgument(!topicName.isEmpty(), "topicName can not be empty");
+    checkIsUsable();
+
+    LOG.info("Creating topic '{}'...", topicName);
+
+    Topic topic = topicAdminClient.createTopic(getTopicName(topicName));
+    TopicName reference = PubsubUtils.toTopicName(topic);
+    createdTopics.add(reference);
+
+    LOG.info("Topic '{}' was created successfully!", topicName);
+
+    return reference;
+  }
+
+  @Override
+  public SubscriptionName createSubscription(String topicName, String subscriptionName) {
+    checkArgument(!topicName.isEmpty(), "topicName can not be empty");
+    checkArgument(!subscriptionName.isEmpty(), "subscriptionName can not be empty");
+    checkIsUsable();
+
+    return createSubscription(getTopicName(topicName), subscriptionName);
+  }
+
+  @Override
+  public SubscriptionName createSubscription(TopicName topicName, String subscriptionName) {
+    checkArgument(!subscriptionName.isEmpty(), "subscriptionName can not be empty");
+
+    LOG.info("Creating subscription '{}' for topic '{}'", subscriptionName, topicName);
+
+    Subscription subscription =
```

Comment:
I think it makes sense to verify that the topic is being tracked before adding the subscription, since this should be keeping everything self-contained (i.e. not adding subscriptions to topics not part of this test). We don't need to have a strict mapping between topics and subscriptions right now, but we should make sure that the subscription is being added to some topic that we're aware of.